src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1779,34 +1837,31 @@ IEnumerable<List<ForeignKeyDefinition>> foreignKeysForAllTargetEntities
         }
 
         /// <summary>
-        /// Fills the table definition with the inferred foreign key metadata
-        /// about the referencing and referenced columns.
+        /// Hydrates the table definition (SourceDefinition) with database foreign key
+        /// metadata that define a relationship's referencing and referenced columns.
         /// </summary>
         /// <param name="dbEntitiesToBePopulatedWithFK">List of database entities
         /// whose definition has to be populated with foreign key information.</param>
         private void FillInferredFkInfo(
             IEnumerable<SourceDefinition> dbEntitiesToBePopulatedWithFK)
         {
-            // For each table definition that has to be populated with the inferred
-            // foreign key information.
             foreach (SourceDefinition sourceDefinition in dbEntitiesToBePopulatedWithFK)
             {
-                // For each source entities, which maps to this table definition
-                // and has a relationship metadata to be filled.
                 foreach ((string sourceEntityName, RelationshipMetadata relationshipData)
                        in sourceDefinition.SourceEntityRelationshipMap)
                 {
-                    // Enumerate all the foreign keys required for all the target entities
-                    // that this source is related to.
+                    // Create ForeignKeyDefinition objects representing the relationships
+                    // between the source entity and each of its defined target entities.
                     foreach ((string targetEntityName, List<ForeignKeyDefinition> fKDefinitionsToTarget) in relationshipData.TargetEntityToFkDefinitionMap)
                     {
-                        // 
-                        // Scenario 1: When a FK constraint is defined between source and target entities in the database.
-                        // In this case, there will be exactly one ForeignKeyDefinition with the right pair of Referencing and Referenced tables. 
-                        // Scenario 2: When no FK constraint is defined between source and target entities, but the relationship fields are configured through config file
-                        // In this case, two entries will be created. 
-                        // First entry: Referencing table: Source entity, Referenced table: Target entity
-                        // Second entry: Referencing table: Target entity, Referenced table: Source entity 
+                        // fkDefinitionsToTarget is a List that is hydrated differently depending
```

Comment:
do you mean this hydration has already happened before calling the GetValidatedFKs call below or is that state after the call is complete?

Reply:
The hydration has already happened before calling the GetValidatedFKs. The comment describes the possible states of `fkDefinitionsToTarget` that `GetValidatedFKs()` needs to handle.